### PR TITLE
feat(Timers): add Timers extension

### PR DESCRIPTION
**Pull request by ThatNerdSquared**
## Description

This PR adds a Timers extension to Raycast, allowing the user to start, stop, and save custom countdown timers.

## Screencast

See `metadata` for screenshots; it's hard to take an effective screencast that doesn't end up being excessively long due to the nature of the extension so I'd recommend a test drive by setting a 1-5 minute timer.

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed in the `media` folder

### CODE REVIEW

**Comment by ThatNerdSquared** [comment]
Thoughts on the chime sound when the timer finishes are appreciated, I wanted a longer one but not sure where to procure it from.

File: extensions/timers/src/manageTimers.tsx [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  };
+
+  const formatTime = (timeInSeconds: number | string) => {
+    const time = new Date(timeInSeconds);
+    time.setSeconds(Number(timeInSeconds));
+    return time.toISOString().substring(11, 19);
+  };
+
+  return (
+    <List isLoading={timers == [] || customTimers == {}}>
```
[the remark this answers is not in the document]
Comment (ThatNerdSquared) [reply extensions/timers/src/manageTimers.tsx:61]: @eddyw it's the "`[] == []` returns `false`" error right? Not the most experienced with React so just wanted to make sure I'm not missing anything else.

[the remark this answers is not in the document]
Comment (eddyw) [reply extensions/timers/src/manageTimers.tsx:61]: @ThatNerdSquared it's both `[] == [] // false` and `{} == {} // false`

[the remark this answers is not in the document]
Comment (ThatNerdSquared) [reply extensions/timers/src/manageTimers.tsx:61]: @eddyw got it, habits from Python haha. Thanks for catching this!

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @ThatNerdSquared 👋 

Thanks for an awesome extension 💪 

I have some suggestions and comments for you to address, feel free to catch me in [Slack](https://raycastcommunity.slack.com/) if you want to discuss.

- When using `Manage Timers` then the loader keeps spinning, both when no timers and with added timers.

- When using the “Save as Preset” feature, it's not remembering the settings, how is it supposed to work? Added “Belly” as preset, but can't find it as in your metadata image.

66Ac7Nq4j3ooI22S
- Would be cool if you redraw the timers once each second, something like that is already added in the [Authy extension](https://github.com/raycast/extensions/tree/main/extensions/authy)

- In the rename screen, I think it would be better to set the current name as placeholder

9prFkX6R6CPSuhQd
- You could maybe get it to read the title of the timer when the time is over using `say` command

- Create a dropdown preference with some difference sounds, you should be able to put a 2-3 seconds optimized file in the assets folder and use the [environment variable](https://developers.raycast.com/api-reference/environment#environment) to get the path

Really looking forward to testing it out again 🥇 

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by ThatNerdSquared** [comment]
@pernielsentikaer "Save as Preset" should save to a `customTimers.json` in the folder defined by `environment.supportPath`, and there is code to create it if it doesn't exist... could you check the extension support folder on your side to see if the file gets created at all?

**Comment by pernielsentikaer** [comment]
No problem at all @ThatNerdSquared, just summon me if you need help, have questions or just want to chat 😄

**Comment by sxn** [comment]
hi @ThatNerdSquared 👋 

I wanted to check what the status on this PR is. Any idea when you'll have time to implement the requested changes? 😸

**Comment by ThatNerdSquared** [comment]
Hi @sxn!

I'm working on some changes locally, but I've been pretty swamped with a few other projects on the side. I'm hoping to get the rest of the changes out by next Thursday, sorry for the delay! 😅

**Comment by sxn** [comment]
alright, take your time! I'll close the PR until then - feel free to reopen it once things calm down a bit 🤗

**Pull request closed without merging**